pkgs/development/compilers/terra/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, lua, fetchFromGitHub, fetchurl, which, llvm, clang, ncurses }:
+
+let luajit_src = fetchurl {
+  url = "http://luajit.org/download/LuaJIT-2.0.4.tar.gz";
+  sha256 = "0zc0y7p6nx1c0pp4nhgbdgjljpfxsb5kgwp4ysz22l1p2bms83v2";
+};
+in stdenv.mkDerivation rec {
+  name = "terra-git-${version}";
+  version = "2016-01-06";
+
+  src = fetchFromGitHub {
+    owner = "zdevito";
+    repo = "terra";
+    rev = "914cb98b8adcd50b2ec8205ef5d6914d3547e281";
+    sha256 = "1q0dm9gkx2lh2d2sfgly6j5nw32qigmlj3phdvjp26bz99cvxq46";
+  };
+
+  patchPhase = ''
+    substituteInPlace Makefile --replace \
+      '-lcurses' '-lncurses'
+  '';
+
+  configurePhase = ''
+    mkdir -p build
+    cp ${luajit_src} build/LuaJIT-2.0.4.tar.gz
```

Comment:
Sorry. I meant:

When bumping `luajit_src`, someone will very likely miss/forget this line, and be mightily confused when the build fails with an error message still mentioning the ‘old’ version.

Seems best to avoid that possibility and get rid of the duplication in one go:

```
let
  luajitArchive = "LuaJIT-2.0.4.tar.gz";
  luajitSrc = fetchurl {
    url = "http://luajit.org/download/${luajitArchive}";
    sha256 = "0zc0y7p6nx1c0pp4nhgbdgjljpfxsb5kgwp4ysz22l1p2bms83v2";
  };
in
# ...lots of stuff...
  configurePhase = ''
    mkdir -p build
    cp ${luajitSrc} build/${luajitArchive}
```